src/black/files.py

```diff
@@ -198,7 +198,7 @@ def gen_python_files(
     extend_exclude: Optional[Pattern[str]],
     force_exclude: Optional[Pattern[str]],
     report: Report,
-    gitignore: Optional[PathSpec],
+    gitignore: Optional[Dict[Path, PathSpec]],
```

Comment:
Would suggest renaming this to `gitignore_dict` for clarity. Also, no reason to keep this Optional, we can just use an empty dict if there's no gitignores.

Reply:
`gitignore` can be `None` if `exclude` is not `None`. In that case, no gitignore is considered at any level. I though that this is the expected behavior while reading the code. If `get_sources` receives a value for `exclude`, what's the expected behavior?